lib/gcloud/datastore/dataset.rb

```diff
@@ -321,6 +319,133 @@ def transaction
         end
       end
 
+      ##
+      # Create a new Query instance. This is a convenience method to make the
+      # creation of Query objects easier.
+      #
+      # === Parameters
+      #
+      # +kinds+::
+      #   The kind of entities to query. This is optional. (+String+)
+      #
+      # === Returns
+      #
+      # Gcloud::Datastore::Query
+      #
+      # === Example
+      #
+      #   query = dataset.query("Task").
+      #     where("completed", "=", true)
+      #   tasks = dataset.run query
+      #
+      # This code is equivalent to the following:
+      #
+      #   query = Gcloud::Datastore::Query.new.
+      #     kind("Task").
+      #     where("completed", "=", true)
+      #   tasks = dataset.run query
+      #
+      def query *kinds
+        query = Query.new
+        query.kind(*kinds) unless kinds.empty?
+        query
+      end
+
+      ##
+      # Create a new Key instance. This is a convenience method to make the
+      # creation of Key objects easier.
+      #
+      # === Parameters
+      #
+      # +kind+::
+      #   The kind of the Key. This is optional. (+String+)
+      # +id_or_name+::
+      #   The id or name of the Key. This is optional. (+Integer+ or +String+)
+      #
+      # === Returns
+      #
+      # Gcloud::Datastore::Key
+      #
+      # === Example
+      #
+      #   key = dataset.key "User", "username"
+      #
+      # This code is equivalent to the following:
+      #
+      #   key = Gcloud::Datastore::Key.new "User", "username"
+      #
+      def key kind = nil, id_or_name = nil
+        Key.new kind, id_or_name
+      end
+
+      ##
+      # Create a new empty Entity instance. This is a convenience method to make
+      # the creation of Entity objects easier.
+      #
+      # === Parameters
+      #
+      # +key_or_kind+::
+      #   A Key object or +kind+ string value. This is optional. (+Key+ or
+      #   +String+ or +nil+)
+      # +id_or_name+::
+      #   The Key's +id+ or +name+ value if a +kind+ was provided in the first
+      #   parameter. (+Integer+ or +String+ or +nil+)
+      #
+      # === Returns
+      #
+      # Gcloud::Datastore::Entity
+      #
+      # === Examples
+      #
+      #   entity = dataset.entity
+      #
+      # This code is equivalent to the following:
+      #
+      #   entity = Gcloud::Datastore::Entity.new
+      #
+      # The key can also be passed in as an object:
+      #
+      #   key = dataset.key "User", "username"
```

Comment:
Totes agree. Added a new commit to replace the vague `"username"` with the concrete `"heidi@example.com"` in the documentation.